app/src/main/java/com/serranoie/app/itinero/ui/MainActivity.kt

```
@@ -95,12 +115,87 @@ class MainActivity : ComponentActivity() {
+
+            // If we get here without exception, the token is valid
+            android.util.Log.d(
+                "ITINERO - MainActivity",
+                "Token validation successful via travel endpoint"
+            )
+
+        } catch (e: UnauthorizedException) {
+            // Token is invalid, this will be handled by UnauthorizedHandler
+            android.util.Log.w("ITINERO - MainActivity", "Token is unauthorized (401)")
+            throw e
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)